src/engine/sparqlExpressions/SparqlExpressionTypes.h

```diff
@@ -127,7 +127,9 @@ CPP_template(typename ResultT)(
       -> CPP_ret(ExpressionResult)(
           requires SingleExpressionResult<std::decay_t<decltype(x)>>) {
     using R = std::decay_t<decltype(x)>;
-    if constexpr (std::is_constructible_v<R, decltype(AD_FWD(x))>) {
+    if constexpr (std::is_constructible_v<R, decltype(AD_FWD(x))> &&
+                  !ad_utility::similarToInstantiation<R,
+                                                      VectorWithMemoryLimit>) {
```

Comment:
Have a look at what is going on here (it has to do with the expression types and the private/publicness of the `clone` method.